### PR TITLE
Setup quality tools and CI

### PR DESCRIPTION
**Pull request by bamorim**
This is re-doing all the CI and quality bootstrap now that we started fresh on 1.7.

- [x] Setup ex_check and related tools
- [x] Setup CI

A few things I noticed from existing setup that I did differently:
- I'm using `mix test --cover` instead of `mix coveralls` on local dev, since we just want to print to console and we already configured coveralls as the test coverage tool.
- I didn't understand why we had two different workflows "Elixir" and "Coveralls". I joined in one CI job that first runs everything but coveralls then proceeds to run coveralls.

### CODE REVIEW

File: .check.exs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[
+  tools: [
+    {:credo, "mix credo --strict"},
```
Comment (rands0n) [inline .check.exs:3]: Don't we need to `-a` flag as well?

File: .credo.exs
Comment (rands0n) [inline .credo.exs:1]: What about enabling our check, it is?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+          ${{ runner.os }}-${{ steps.beam.outputs.elixir-version }}-${{ steps.beam.outputs.otp-version }}-plt
+        path: priv/plts
+
+    # Create PLTs if no cache was found
+    - name: Create PLTs
+      if: steps.plt_cache.outputs.cache-hit != 'true'
+      run: mix dialyzer --plt
+
+    - name: Run check
+      run: mix check --except ex_unit
```
Comment (rands0n) [inline .github/workflows/ci.yaml:110]: Why remove `ex_unit` here?

**Review by rands0n — APPROVED** [review]

File: .check.exs (hunk 1 above)
Comment (bamorim) [reply .check.exs:3]: I have no idea what that does.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (bamorim) [reply .github/workflows/ci.yaml:110]: Because we'll run the tests afterwards using coveralls.

File: .credo.exs
Comment (bamorim) [reply .credo.exs:1]: I want to review all the checks we want later, but need more time. For now I went with the default config.

**Comment by rands0n** [comment]
> I'm using mix test --cover instead of mix coveralls on local dev, since we just want to print to console and we already configured coveralls as the test coverage tool.

Perfection, I'm also using it in a private project.

**Pull request merged**